paramiko/proxy.py

```diff
@@ -119,7 +118,11 @@ def recv(self, size):
             raise ProxyCommandFailure(" ".join(self.cmd), e.strerror)
 
     def close(self):
-        os.kill(self.process.pid, signal.SIGTERM)
+        self.process.stdin.close()
+        self.process.stdout.close()
+        self.process.stderr.close()
+        self.process.terminate()
+        self.process.wait()
```

Comment:
Is this `.wait()` a good idea? Does `os.kill()` wait for the process to close? I can't tell from the [docs](https://docs.python.org/3/library/os.html#os.kill).

I'm concerned this could lead to hangs if for some reason a process fails to terminate cleanly.